src/backend/didrequest.c

```diff
@@ -267,7 +270,12 @@ DIDDocument *DIDRequest_FromJson(DIDRequest *request, cJSON *json)
         DIDError_Set(DIDERR_RESOLVE_ERROR, "Invalid payload.");
         return NULL;
     }
-    request->payload = strdup(cJSON_GetStringValue(item));
+    char *payload = cJSON_GetStringValue(item);
```

Comment:
please try  to follow the rule of declaration variables at the entry of scope of effect, like at the entrance of function.